examples/apply/apply_bp/apply_bp.jl

```diff
@@ -92,29 +89,28 @@ function simple_update_bp(
 
       # Update message tensor
       ψψ = norm_network(ψ)
-      mts[s1] = ITensorNetwork(dictionary([(v1, 1) => ψψ[v1, 1], (v1, 2) => ψψ[v1, 2]]))
-      mts[s2] = ITensorNetwork(dictionary([(v2, 1) => ψψ[v2, 1], (v2, 2) => ψψ[v2, 2]]))
-      mts[s1 => s2] = ITensorNetwork(obs.singular_values)
-      mts[s2 => s1] = ITensorNetwork(obs.singular_values)
+      pψψ = PartitionedGraph(ψψ, group(v -> v[1], vertices(ψψ)))
+      mts[pe] = dense.(obs.singular_values)
+      mts[reverse(pe)] = dense.(obs.singular_values)
     end
     if regauge
       println("regauge")
       mts = belief_propagation(
-        ψψ, mts; contract_kwargs=(; alg="exact"), niters=50, target_precision=1e-5
+        pψψ, mts; contract_kwargs=(; alg="exact"), niters=50, target_precision=1e-5
       )
     end
   end
-  return ψ, mts
+  return ψ, pψψ, mts
 end
 
 function simple_update_vidal(os, ψ::ITensorNetwork; maxdim, regauge=false)
   println("Simple update, Vidal gauge")
   ψψ = norm_network(ψ)
-  mts = message_tensors(partition(ψψ, group(v -> v[1], vertices(ψψ))))
+  pψψ = PartitionedGraph(ψψ, group(v -> v[1], vertices(ψψ)))
   mts = belief_propagation(
-    ψψ, mts; contract_kwargs=(; alg="exact"), niters=50, target_precision=1e-5
+    pψψ; contract_kwargs=(; alg="exact"), niters=50, target_precision=1e-5
   )
-  ψ, bond_tensors = vidal_gauge(ψ, mts)
+  ψ, bond_tensors = vidal_gauge(ψ, pψψ, mts)
```

Comment:
Again, it feel strange to me to now have to pass `ψ`, `pψψ`, and `mts`. Let's discuss this, my guess is there is a better design.